### ISSUE TITLE
Test stops responding when first "// AfterAll" displayed

### ISSUE THREAD

**Issue by JohnMasen** [dotnet/BenchmarkDotNet#1383]
I found my test will stop responding on first "//AfterALL" output. the following clues during debugging.
1. Testing on GPU accelerated library will trigger this issue, CPU library does not have this problem
2. the problem will gone if I use "InProcessEmitToolchain", however it will trigger  [GlobalSetup] method crash when more than 1 parameters are used. for example
```
[Params(10000)]  // it works
 public int Items { get; set; } 
```

```
[Params(1000,10000)] //[GlobalSetup] method will crash when Items=10000 
public int Items { get; set; } 
```
3. tried to manual run the test code to figure out if the test code hang it self, turns out test code is fine.
```
ComputeShaderPerformanceILGPU c = new ComputeShaderPerformanceILGPU() { CopyBack = false, Items = 1000 };
c.Acc = c.GetAccelerators.First();
c.Init(); // global setup
c.Step(); //benchmark
c.Close(); //global cleanup
```
Source code is at https://github.com/JohnMasen/VirusSimulator under DEV branch, project "ComputeShaderTest.csproj"

**Comment by adamsitnik** [comment]
Hi @JohnMasen 

I've tried to repo the problem, but I've failed with the following error:

```log
// Execute: dotnet "15dd49e3-2702-44cc-bc2a-74b3c679f3e7.dll" --benchmarkName "ComputeShaderTest.ComputeShaderPerformanceILGPU.Step(Items: 1000, Acc: CPUAccelerator [WarpSize: 1, MaxNumThreadsPerGroup: 12, MemorySize: 9223372036854775807])" --job "Dry" --benchmarkId 0 in C:\Projects\forks\VirusSimulator\src\ComputeShaderTest\bin\Release\netcoreapp3.1\15dd49e3-2702-44cc-bc2a-74b3c679f3e7\bin\Release\netcoreapp3.1
// BeforeAnythingElse

System.TypeLoadException: Could not load type 'ILGPU.Index' from assembly 'ILGPU, Version=0.9.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Reflection.RuntimeAssembly.GetType(QCallAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type, ObjectHandleOnStack keepAlive, ObjectHandleOnStack assemblyLoadContext)
   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)
   at System.Reflection.Assembly.GetType(String name)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\Projects\forks\VirusSimulator\src\ComputeShaderTest\bin\Release\netcoreapp3.1\15dd49e3-2702-44cc-bc2a-74b3c679f3e7\15dd49e3-2702-44cc-bc2a-74b3c679f3e7.notcs:line 48
```

**Comment by adamsitnik** [comment]
Ok, I was able to repro

**Comment by adamsitnik** [comment]
It looks like the following code is keeping the process alive:

![obraz](https://user-images.githubusercontent.com/6011991/96892258-c26e0f80-1489-11eb-99bb-8055987b8a71.png)

![obraz](https://user-images.githubusercontent.com/6011991/96892319-d1ed5880-1489-11eb-8767-34dfc46850b6.png)

**Comment by adamsitnik** [comment]
@JohnMasen this was an insane combination of a few bugs, I've sent a fix with explanation in #1571

**Comment by rdcm** [comment]
@adamsitnik Hi! Look like I have a similar issue with my benchmark. After printed section  `// AfterAll` execution stopped. Could you suggest any workaround while `v0.13.0` not released?

`net5.0`
`BenchmarkDotNet - 0.12.1`
